packages/driver/src/dom/selection.ts

```diff
@@ -563,20 +556,44 @@ const _moveSelectionTo = function (toStart: boolean, el: HTMLElement, options =
   }
 
   if ($elements.isContentEditable(el)) {
-    $elements.callNativeMethod(doc, 'execCommand', 'selectAll', false, null)
-    const selection = doc.getSelection()
+    const selection = _getSelectionByEl(el)
 
     if (!selection) {
       return
     }
 
-    // collapsing the range doesn't work on input/textareas, since the range contains more than the input element
-    // However, IE can always* set selection range, so only modern browsers (with the selection API) will need this
-    const direction = toStart ? 'backward' : 'forward'
+    if (Cypress.isBrowser({ family: 'firefox' })) {
+      // FireFox doesn't treat a selectall+arrow the same as clicking the start/end of a contenteditable
+      // so we need to select the specific nodes inside the contenteditable.
+      const root = getHostContenteditable(el)
 
-    selection.modify('move', direction, 'line')
+      let elToSelect = root.childNodes[toStart ? 0 : root.childNodes.length - 1]
 
-    return
+      // in firefox, an empty contenteditable my be a single <br> element, or <div><br/></div>
+      // but its innerText will be '\n' (maybe find a more efficient measure)
+      if (!elToSelect || root.innerText === '\n') {
+        // we must be in an empty contenteditable, so we're already at both the start and end
+        return
+      }
+
+      // if we're on a <br> but the text isn't empty, we need to
+      if ($elements.getTagName(elToSelect) === 'br') {
+        if (root.childNodes.length < 2) {
+          // no other node to target, shouldn't really happen but we should behave like the contenteditable is empty
+          return
+        }
+
+        elToSelect = toStart ? root.childNodes[1] : root.childNodes[root.childNodes.length - 2]
+      }
+
+      const range = selection.getRangeAt(0)
+
+      range.selectNodeContents(elToSelect)
+    } else {
+      $elements.callNativeMethod(doc, 'execCommand', 'selectAll', false, null)
+    }
+
+    toStart ? selection.collapseToStart() : selection.collapseToEnd()
   }
 
   return false
```

Comment:
We don't need this line. Every `return` above returns `undefined` but it's `false` here. And `moveSelectionToEnd` and `moveSelectionToStart` doesn't need a return value.